curvefs/src/mds/heartbeat/topo_updater.cpp

```diff
@@ -222,6 +222,28 @@ void TopoUpdater::UpdatePartitionTopo(
             LOG(WARNING) << "hearbeat report partition which is not in topo"
                          << ", copysetId = " << copySetId
                          << ", partitionId = " << it.GetPartitionId();
+            
```

Comment:
It's not a good idea to delete here. If you really want to do it, you'd better check inode and dentry numbers of this partition again and make sure it is empty.
This piece of code logic can directly call TopologyManager::DeletePartition.

Reply:
I noticed that the logic in TopologyManager::DeletePartitio first fetches the corresponding partition from topology_. The situation we are facing now is that there is no corresponding partition in topology_.So, I added the DeleteAbnormalPartition function.

Reply:
Failure to create is just one of the cases where the partition exists in metaserver but not in mds, so to be on the safe side, it is best to check whether the partition is empty first.

You can split the original TopologyManager::DeletePartitio into a function, such as TopologyManager::DeletePartitionDirectly, which does not check whether it exists in mds, so TopologyManager::DeletePartitionDirectly can be reused by you and the original code at the same time, and you can There is no need to rewrite the sending logic again.